gfxr_ext/decode/dive_annotation_processor.cpp

```diff
@@ -56,14 +56,38 @@ void DiveAnnotationProcessor::WriteBlockEnd(const gfxrecon::util::DiveFunctionDa
         {
             uint64_t cmd_handle = args["commandBuffer"];
 
-            // There could be cases where vkBeginCommandBuffer is called after vkEndCommandBuffer
-            // without having the command buffer submitted
-            if (vkCmd.name == "vkBeginCommandBuffer")
+            if (vkCmd.name.find("vkBeginCommandBuffer") != std::string::npos)
             {
                 m_cmd_vk_commands_cache[cmd_handle].clear();
+
+                // Total draw count for this command buffer
+                m_draw_call_counts_map[cmd_handle].begin_command_buffer_draw_call_count = 0;
+            }
+            else if (vkCmd.name.find("vkCmdBeginRenderPass") != std::string::npos)
+            {
+                m_render_pass_draw_call_counts_stack[cmd_handle].push(0);
+            }
+            else if (vkCmd.name.find("vkCmdEndRenderPass") != std::string::npos)
+            {
+                if (!m_render_pass_draw_call_counts_stack[cmd_handle].empty())
+                {
+                    uint64_t rp_draw_count = m_render_pass_draw_call_counts_stack[cmd_handle].top();
```

Comment:
out of curiosity, why do you need a stack here? renderpass cannot be nested.

Reply:
Ah ok, removed the use of the stack and use the renderpass draw count vector directly instead.